### PR TITLE
chore: update golang.org/x/*

### PR DESCRIPTION
**Pull request by karlkfi**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -160,12 +159,13 @@ require (
 	go.starlark.net v0.0.0-20230525235612-a134d8f9ddca // indirect
 	go.uber.org/atomic v1.11.0 // indirect
 	golang.org/x/crypto v0.31.0 // indirect
+	golang.org/x/oauth2 v0.24.0 // indirect
```
[the remark this answers is not in the document]
Comment (karlkfi) [reply go.mod:162]: it was just in the wrong place. moved down to the indirect section

**Review by sdowell — APPROVED** [review]
/lgtm

File: go.mod [hunk 2]
```diff
@@ -35,10 +35,9 @@ require (
 	go.opencensus.io v0.24.0
 	go.uber.org/multierr v1.11.0
 	go.uber.org/zap v1.26.0
-	golang.org/x/exp v0.0.0-20240506185415-9bf2ced13842
-	golang.org/x/mod v0.18.0
-	golang.org/x/net v0.29.0
-	golang.org/x/oauth2 v0.24.0 // indirect
+	golang.org/x/exp v0.0.0-20241217172543-b2144cdd0a67
+	golang.org/x/mod v0.22.0
+	golang.org/x/net v0.33.0
```
[the remark this answers is not in the document]
Comment (karlkfi) [reply go.mod]: I removed golang.org/x/net, but it's not clear how to remove the others.

[the remark this answers is not in the document]
Comment (karlkfi) [reply go.mod]: Figured it out. It's just "maps" and "slices".

**Review by janetkuo — APPROVED** [review]
/lgtm

**Comment by karlkfi** [comment]
Reverted the "golang.org/x/exp/maps" change to avoid needing to update Go to 1.23 yet

**Comment by karlkfi** [comment]
`TestFunnel_Start/RetryEvents_and_SyncEvents_with_RetryBackoff_longer_than_SyncPeriod` appears to be flaky. but there's nothing in this PR that should affect it.

```
I1220 00:14:47.274764   19270 funnel.go:39] Starting event loop
I1220 00:14:47.274959   19270 funnel.go:44] Starting event source: SyncEvent
I1220 00:14:47.275197   19270 funnel.go:76] Stopping event loop: context canceled
I1220 00:14:47.275490   19270 funnel.go:39] Starting event loop
I1220 00:14:47.275537   19270 funnel.go:44] Starting event source: StatusUpdateEvent
I1220 00:14:47.275698   19270 funnel.go:76] Stopping event loop: context canceled
I1220 00:14:47.275927   19270 funnel.go:39] Starting event loop
I1220 00:14:47.275982   19270 funnel.go:44] Starting event source: NamespaceSyncEvent
I1220 00:14:47.277484   19270 funnel.go:76] Stopping event loop: context canceled
I1220 00:14:47.280272   19270 funnel.go:39] Starting event loop
I1220 00:14:47.280331   19270 funnel.go:44] Starting event source: RetrySyncEvent
I1220 00:14:47.280503   19270 funnel.go:76] Stopping event loop: context canceled
I1220 00:14:47.280647   19270 funnel.go:39] Starting event loop
I1220 00:14:47.280654   19270 funnel.go:44] Starting event source: RetrySyncEvent
I1220 00:14:47.280764   19270 funnel.go:76] Stopping event loop: context canceled
I1220 00:14:47.281316   19270 funnel.go:39] Starting event loop
I1220 00:14:47.281351   19270 funnel.go:44] Starting event source: SyncEvent
I1220 00:14:47.281386   19270 funnel.go:44] Starting event source: RetrySyncEvent
I1220 00:14:47.281621   19270 publisher.go:[157](https://oss.gprow.dev/view/gs/oss-prow-build-kpt-config-sync/pr-logs/pull/GoogleContainerTools_kpt-config-sync/1516/kpt-config-sync-presubmit/1869898280792494080#1:build-log.txt%3A157)] Retry limit (3) has been reached
I1220 00:14:47.282332   19270 funnel.go:76] Stopping event loop: context canceled
I1220 00:14:47.282843   19270 funnel.go:39] Starting event loop
I1220 00:14:47.282853   19270 funnel.go:44] Starting event source: SyncEvent
I1220 00:14:47.282865   19270 funnel.go:44] Starting event source: RetrySyncEvent
I1220 00:14:52.283[158](https://oss.gprow.dev/view/gs/oss-prow-build-kpt-config-sync/pr-logs/pull/GoogleContainerTools_kpt-config-sync/1516/kpt-config-sync-presubmit/1869898280792494080#1:build-log.txt%3A158)   19270 funnel.go:76] Stopping event loop: context deadline exceeded
I1220 00:14:52.284073   19270 funnel.go:39] Starting event loop
I1220 00:14:52.284091   19270 funnel.go:44] Starting event source: SyncEvent
I1220 00:14:52.284104   19270 funnel.go:44] Starting event source: StatusUpdateEvent
I1220 00:14:52.286255   19270 funnel.go:76] Stopping event loop: context canceled
--- FAIL: TestFunnel_Start (5.01s)
    --- FAIL: TestFunnel_Start/RetryEvents_and_SyncEvents_with_RetryBackoff_longer_than_SyncPeriod (5.00s)
        funnel_test.go:391: Step 1000
        funnel_test.go:448: Received event: events.Event{Type:"RetrySyncEvent"}
        funnel_test.go:391: Step 3000
        funnel_test.go:429: 
            	Error Trace:	/go/src/kpt.dev/configsync/vendor/sigs.k8s.io/cli-utils/pkg/testutil/asserter.go:51
            	            				/go/src/kpt.dev/configsync/vendor/sigs.k8s.io/cli-utils/pkg/testutil/asserter.go:59
            	            				/go/src/kpt.dev/configsync/pkg/parse/events/funnel_test.go:429
            	Error:      	
            	            	Expected
            	            	    <[]events.Event | len:1, cap:1>: [
            	            	        {
            	            	            Type: "RetrySyncEvent",
            	            	        },
            	            	    ]
            	            	to deeply equal
            	            	    <[]events.Event | len:10, cap:16>: [
            	            	        {
            	            	            Type: "RetrySyncEvent",
            	            	        },
            	            	        {
            	            	            Type: "RetrySyncEvent",
            	            	        },
            	            	        {
            	            	            Type: "RetrySyncEvent",
            	            	        },
            	            	        {
            	            	            Type: "RetrySyncEvent",
            	            	        },
            	            	        {Type: "SyncEvent"},
            	            	        {
            	            	            Type: "RetrySyncEvent",
            	            	        },
            	            	        {Type: "SyncEvent"},
            	            	        {Type: "SyncEvent"},
            	            	        {Type: "SyncEvent"},
            	            	        {
            	            	            Type: "RetrySyncEvent",
            	            	        },
            	            	    ]
            	            	Diff (- Expected, + Actual):
            	            	      []events.Event{
            	            	    - 	{Type: "RetrySyncEvent"},
            	            	    - 	{Type: "RetrySyncEvent"},
            	            	    - 	{Type: "RetrySyncEvent"},
            	            	    - 	{Type: "RetrySyncEvent"},
            	            	    - 	{Type: "SyncEvent"},
            	            	    - 	{Type: "RetrySyncEvent"},
            	            	    - 	{Type: "SyncEvent"},
            	            	    - 	{Type: "SyncEvent"},
            	            	    - 	{Type: "SyncEvent"},
            	            	      	{Type: "RetrySyncEvent"},
            	            	      }
            	            	    
            	Test:       	TestFunnel_Start/RetryEvents_and_SyncEvents_with_RetryBackoff_longer_than_SyncPeriod
FAIL
FAIL	kpt.dev/configsync/pkg/parse/events	5.134s
```

**Review by sdowell — APPROVED** [review]
/lgtm

**Pull request merged**